test/unit/specs/components/DataCollector/index.spec.js

```diff
@@ -143,9 +143,14 @@ describe("Event Command", () => {
     });
   });
 
-  it("Calls onBeforeDataCollection", () => {
+  it("Calls onBeforeDataCollection with the requestId", () => {
     return eventCommand({}).then(() => {
+      expect(sendRequestSpy).toHaveBeenCalled();
       expect(onBeforeDataCollectionSpy).toHaveBeenCalled();
```

Comment:
I don't think this line provides value anymore since you have a `expect(onBeforeDataCollectionSpy).toHaveBeenCalledWith` below.